server/src/main/java/org/elasticsearch/search/fetch/subphase/highlight/HighlightUtils.java

```diff
@@ -46,24 +43,10 @@ private HighlightUtils() {
      * Load field values for highlighting.
      */
     public static List<Object> loadFieldValues(MappedFieldType fieldType,
-                                               FetchSubPhase.HitContext hitContext,
-                                               boolean forceSource) throws IOException {
-        //percolator needs to always load from source, thus it sets the global force source to true
-        List<Object> textsToHighlight;
-        if (forceSource == false && fieldType.isStored()) {
-            CustomFieldsVisitor fieldVisitor = new CustomFieldsVisitor(singleton(fieldType.name()), false);
-            hitContext.reader().document(hitContext.docId(), fieldVisitor);
-            textsToHighlight = fieldVisitor.fields().get(fieldType.name());
-            if (textsToHighlight == null) {
-                // Can happen if the document doesn't have the field to highlight
-                textsToHighlight = Collections.emptyList();
-            }
-        } else {
-            SourceLookup sourceLookup = hitContext.sourceLookup();
-            textsToHighlight = sourceLookup.extractRawValues(fieldType.name());
-        }
-        assert textsToHighlight != null;
-        return textsToHighlight;
+                                               MapperService mapperService,
+                                               FetchSubPhase.HitContext hitContext) throws IOException {
+        ValueFetcher fetcher = fieldType.valueFetcher(mapperService, null, null);
```

Comment:
`null` looks is going to cause this to fail on runtime fields. Do we filter those out other places?

Reply:
I meant to highlight this with a comment of 'not sure about this' - I'm pretty sure that we filter out non-text fields further up the stack, but I need to double-check.

Reply:
I think we *do* try to highlight runtime fields. I'm not sure how we manage not to fail here. Runtime fields really need the `SearchLookup` to do anything.

Reply:
I vaguely remember looking at this too, and finding that for runtime fields we simply don't return anything highlighted rather than failing, as they are not in _source nor stored.

Reply:
Given that runtime fields can't be highlighted anyway, I just added a null check for SearchLookup inside the value fetcher in AbstractScriptFieldType.